codegen/generate.go

```diff
@@ -58,23 +62,39 @@ type fileCreator struct {
 // GenerateCode is exported because it's the only method intended to be used by
 // other objects. Unexported methods may be available to other code in this package,
 // but they're not intended to be used by anything but the fileCreator.
-func (c *fileCreator) GenerateCode(endpoint string, endpointInfo *framework.OASPathItem, tmplType templateType) error {
-	pathToFile, err := codeFilePath(tmplType, endpoint)
+func (c *fileCreator) GenerateCode(endpoint string, endpointInfo *framework.OASPathItem, addedInfo *additionalInfo) error {
```

Comment:
Same comment as above - this would be a big departure from all the code I've seen so far at Vault, so let's hold on this until there's been a team discussion.

Reply:
In looking at the criteria [here](https://github.com/golang/go/wiki/CodeReviewComments#receiver-type), I think it should remain a pointer, mainly because:

>If the receiver is a large struct or array, a pointer receiver is more efficient. How large is large? Assume it's equivalent to passing all its elements as arguments to the method. If that feels too large, it's also too large for the receiver.

In the method `fCreator.GenerateCode(endpoint, paths[endpoint], addedInfo)`, if we added 2 more arguments, the logger and template handler, that would feel large to me because it'd be 5 arguments, and we'd be forced to wrap the line. 

However, the word "feels" is by nature still rather subjective, and the guidance there is also:

>Finally, when in doubt, use a pointer receiver.